src/OMSimulatorLib/System.cpp

```diff
@@ -1604,7 +1604,15 @@ oms_status_enu_t oms::System::updateDependencyGraphs()
       if (validConnection)
       {
         initialUnknownsGraph.addEdge(Connector(varA->getCausality(), varA->getType(), connection->getSignalA()), Connector(varB->getCausality(), varB->getType(), connection->getSignalB()));
-        outputsGraph.addEdge(Connector(varA->getCausality(), varA->getType(), connection->getSignalA()), Connector(varB->getCausality(), varB->getType(), connection->getSignalB()));
+
+        /*
+         * Remove top System level Parameters Dependencies during Simulation, as it is processed during initialization
+         * check for (TopLevel System and not parameter) (or) allow all element level connections
+         */
+        if ((connection->getSignalA().isValidIdent() && !varA->isParameter()) || !connection->getSignalA().isValidIdent())
```

Comment:
@arun3688 Why do you check for `connection->getSignalA().isValidIdent()`? Shouldn't it be enough to check for `variability != parameter?`

Reply:
yes we can check for only for variability != parameter?, see https://github.com/OpenModelica/OMSimulator/pull/757/commits/03b1c1e72c3deb529eb27083f9439ab06f9fe157